kore/test/Test/Kore/Step/SMT/Translate.hs

```diff
@@ -158,6 +158,18 @@ test_translatePredicateWith =
             & expectJustT
             & Test.SMT.runNoSMT
         assertEqual "" actual1 actual2
+    , testCase "b = a, both constructors" $
+            translating (peq Mock.b Mock.a)
+        `yields`
+            (var 0 `eq` var 1)
+    , testCase "f() = a, f functional, a constructor" $
+            translating (peq Mock.functional00 Mock.a)
+        `yields`
+            (var 0 `eq` var 1)
```

Comment:
I wasn't able to get the constructors to not get translated as `var n`. I tried adding a `translatingSMT` which would use the solver with the `MockSymbols` environment, to which I added a constructor declaration for `a`. Anyway, I think that it's not that important, as this successfully tests that `\equals a b` gets translated correctly even if `translateApplication` fails.

Reply:
I wouldn't expect `translatingSMT` to help because we don't need access to the actual solver for this. I investigated further, and it turns out that the SMT declarations builder requires constructors to be declared twice: once as part of their sort and once as symbols. Could you please add a comment to this test, saying that it doesn't return the correct result for this reason? Also, please remove the `smtConstructor` parts from `MockSymbols`, because that function doesn't do anything.

Reply:
Thank you for looking into this!